### PR TITLE
fix: no corresponding SFC entry in the cache

### PR DESCRIPTION
**Pull request by inkyMountain**
A reproduction can be viewed via:
[https://github.com/inkyMountain/vue2-vite](https://github.com/inkyMountain/vue2-vite)

Reproduction steps:
```sh
npm i
npm run dev
open http://localhost:3000
```

When will this bug occur?
Assuming we have a sfc named App.vue with a script block using src attribute like this:
```vue
<template>
<!--some html here-->
</template>

<script src="./appScript.ts">
</script>
```

```ts
// appScript.ts
import {add} from './utils'

console.log(add(1, 2))
export default {}
```

```ts
// utils.ts
export const add = (a: number, b: number) => a + b
```

The request for utils.ts will cause a bug cause its url looks like:
/Users/chenyitao/dev/demos/vue2-vite/src/appScript.ts?vue&type=script&src&from=/Users/chenyitao/dev/demos/vue2-vite/src/App.vue&lang.ts

It's the request for utils.ts but end up with appScript.ts. The reason for this bug is that the from query contains slashes ( / ). The import statement in `appScript.ts` `import {add} from './utils'` will be transformed to:
```ts
import {add} from '/Users/chenyitao/dev/demos/vue2-vite/src/appScript.ts?vue&type=script&src&from=/Users/chenyitao/dev/demos/vue2-vite/src/App.vue&lang.ts'
```

The transforming works this way:
1. vite-plugin-vue2 skips the transform for `appScript.ts` because its type is script instead of style or template.
2. So vite:analysis-plugin takes over the transform.
3. vite:import-analysis plugin call resolve method, so plugin container does a resolve action.
4. vite:resolve plugin starts resolving, and one line of this plugin's code use path.dirname. The slashes in the query influence the result of path.dirname, which should transition from  `/foo/bar?from=/Users/foo/bar` to `/foo?from=/Users/foo/bar`, but actually transition from `/foo/bar?from=/Users/foo/bar` to `/foo/bar?from=/Users/foo`.

A description in Chinese below.
`appScript.ts`中 `utils.ts` 的引入路径 被 vite:resolve 插件错误地转换了，因为`appScript.ts`的请求路径中包含了from参数，参数的值里有 / ，但是没有被转译。vite:resolve 插件内部使用了path.dirname，这个方法被from参数中的 / 影响，输出了错误的结果。可以看一下我最上面贴的复现repo。

### CODE REVIEW

**Comment by bazz** [comment]
Just encountered same issue. Any plans to merge this fix?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "vite-plugin-vue2",
-  "version": "1.9.0",
```
Comment (antfu) [inline package.json:3]: Can you restore this changes?

**Review by antfu — CHANGES_REQUESTED** [review]
Please revert
- Changelog
- Version and naming changes

File: package.json (hunk 1 above)
Comment (inkyMountain) [reply package.json]: @antfu 
Hello there, I did revert commit and there's only necessary file changes now. Please review again thx.

**Review by antfu — APPROVED** [review]

**Pull request merged**